src/hooks.js

```diff
@@ -0,0 +1,38 @@
+/** @typedef {import("webpack").Compiler} WebpackCompiler */
+/**
+ * @typedef {Object} MiniCssExtractHooks
+ * @property {import("tapable").SyncBailHook<{ varNames: { tag: string, chunkId: string } }, string>} [beforeTagInsert]
+ */
+
+const { SyncBailHook } = require("tapable");
+
+/**
+ * @type {WeakMap<WebpackCompiler, MiniCssExtractHooks>}}
+ */
+const miniCssExtractHooksMap = new WeakMap();
+
+/**
+ * @param {WebpackCompiler} complier
+ */
+function getHooks(complier) {
+  let hooks = miniCssExtractHooksMap.get(complier);
+  // Setup the hooks only once
+  if (!hooks) {
+    hooks = createHooks();
+    miniCssExtractHooksMap.set(complier, hooks);
+  }
+  return hooks;
+}
+
+/**
+ * @returns {MiniCssExtractHooks}
+ */
+function createHooks() {
+  return {
+    beforeTagInsert: new SyncBailHook(["vars"]),
+  };
+}
+
+module.exports = {
+  getHooks,
+};
```

Comment:
Let's make it static methods too

Reply:
> Let's make it static methods too

getHooks is already a static method in the MiniCssExtractPlugin class, is there anything I need to change here?

Reply:
Let's use this pattern https://github.com/webpack/webpack/blob/main/lib/CleanPlugin.js#L296, also we should use compilation instead compiler

Reply:
> Let's use this pattern https://github.com/webpack/webpack/blob/main/lib/CleanPlugin.js#L296, also we should use compilation instead compiler

done